src/celeritas/field/RZPhiMapFieldData.hh

```diff
@@ -0,0 +1,133 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/field/RZPhiMapFieldData.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/cont/Array.hh"
+#include "corecel/data/Collection.hh"
+#include "corecel/data/HyperslabIndexer.hh"
+#include "corecel/math/Turn.hh"
+
+#include "FieldDriverOptions.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * FieldMap (3-dimensional RZ-Phi map) grid data
+ */
+template<Ownership W, MemSpace M>
+struct RZPhiMapGridData
+{
+    template<class T>
+    using Items = Collection<T, W, M>;
+    Items<real_type> storage;  //!< [Phi, R, Z]
+    Array<size_type, 3> grid_size;  //!< [Phi, R, Z]
+
+    ItemRange<real_type> phi;  //!< Index range for phi
+    ItemRange<real_type> r;  //!< Index range for r
+    ItemRange<real_type> z;  //!< Index range for z
+
+    //! Check whether the data is assigned
+    explicit inline CELER_FUNCTION operator bool() const
+    {
+        return !storage.empty() && grid_size[0] > 1 && grid_size[1] > 1
+               && grid_size[2] > 1 && phi.size() == grid_size[0]
+               && r.size() == grid_size[1] && z.size() == grid_size[2];
```

Comment:
I wish the `HyperslabIndexer` didn't require creating an extra array for the dimensions... because of that it often looks cleaner/clearer/requires less code just to write out the index calculation (@sethrj).

Reply:
should we order it as R-Phi-Z, which is the standard ordering?

Reply:
The "standard" ordering is, based on my experience, a holdover from fortran indexing. We have the chance to start clean and we should take it :)

Reply:
should I rename the map `PhiRZ` then?

Reply:
Oh I didn't notice the ordering was flipped around from the name 😭 Maybe we should just call it CylFieldMap... and the RZ one SimpleCylFieldMap (because it's both the field vector and the grid that are azimuthally symmetric)? 

Let's order it naturally: `[x][y][z]` if it's cartesian, `[r][phi/theta][z]` or `[r][z]` if cylindrical, `[r][theta][phi]` if polar.

In my old code base we'd order a cartesian grid `data[z][y][x]` so it would have the same stride pattern as a fortran `data(x, y, z)` grid. I think our `RZField`, which is based on CMS data, uses `[z][r]` for the same reason (and I'd like to change it before v1).